pkg/resourcemapping/operator_resource_mapping.go

```diff
@@ -321,102 +321,92 @@ func (ormClient *ORMClient) LocateOwnerPaths(ownedObj *unstructured.Unstructured
-	glog.Infof("Update owner %s/%s resources found using orm v1 for source %s/%s",
-		controllerOwnerReference.Kind, controllerOwnerReference.Name,
-		updatedControllerObj.GetKind(), updatedControllerObj.GetName())
-
-	operatorResourceSpec, exists := ormClient.operatorResourceSpecMap[operatorCRUID]
-	if !exists {
-		return fmt.Errorf("operatorResourceSpec not found in operatorResourceSpecMap for operatorCR %s", operatorCRUID)
-	}
-	ormTemplate, exists := operatorResourceSpec.ormTemplateMap[componentKey]
-	if !exists {
-		return fmt.Errorf("ormTemplate not found in ormTemplateMap for componentKey %s for operatorCR %s", componentKey, operatorCRUID)
-	}
-
-	operatorResKind := controllerOwnerReference.Kind //operator kind and instance
-	operatorResName := controllerOwnerReference.Name
-	operatorRes := operatorResKind + "/" + operatorResName
-
-	resourceNamespace := updatedControllerObj.GetNamespace()
-	if operatorResourceSpec.isClusterScope {
-		resourceNamespace = v1.NamespaceAll
-	}
-	dynResourceClient := ormClient.dynClient.Resource(operatorResourceSpec.operatorGVResource).Namespace(resourceNamespace)
-	operatorCR, err := dynResourceClient.Get(context.TODO(), operatorResName, metav1.GetOptions{})
-	if err != nil {
-		return fmt.Errorf("failed to get CR %s for %s in namespace %s: %v", operatorRes, componentKey, resourceNamespace, err)
-	}
-
-	glog.V(2).Infof("Updating CR %s for %s in namespace %s", operatorRes, componentKey, resourceNamespace)
+// controllerOwnerReference -- ownerReference of a K8s controller, which contains metadata of a owner CR
+// ownerResources -- mapping of owner resources with owner resource path and source/owned controller obj
+func (ormClient *ORMClientManager) UpdateOwners(updatedControllerObj *unstructured.Unstructured, controllerOwnerReference discoveryutil.OwnerInfo, ownerResources *OwnerResources) error {
+	var ownerCR *unstructured.Unstructured
+	var dynResourceClient dynamic.ResourceInterface
+	var err error
+	isORMV1 := len(ormClient.ORMClient.operatorResourceSpecMap) != 0
 	updated := false
-	// Update based on each resourceMappingTemplate
-	for _, resourceMappingTemplate := range ormTemplate.resourceMappingTemplates {
-		// Set resourceMappingComponentName to resourceMappingTemplate so as to parse the srcPath and destPath based on text template
-		resourceMappingTemplate[resourceMappingComponentName] = ormTemplate.componentName
-		srcPath, destPath, err := ormClient.parseSrcAndDestPath(resourceMappingTemplate)
+	// for V1 orm client it builds operator resource mapping templates fetched from OperatorResourceMapping CR in discovery client
+	// so if this cached map data from Operator-managed CustomResource UID to ORMSpec is populated, meaning it discovred V1 ORM.
+	// The reason for getting V1 owner CR and V2 owner CR seperately is the GVR resources from owner resources is different for V2 comapred
+	// to V1, for V1 we get it from operator resource spec which is owner orm itself
+	if isORMV1 {
```

Comment:
If the ownerCR is handled by V1, I'd suggest to use the original code to process it. And when we drop v1, we drop those code all together.